src/streaming/text/TextTracks.js

```diff
@@ -657,24 +679,59 @@ function TextTracks(config) {
         return cue;
     }
 
+    function _getCueInformation(currentItem, timeOffset) {
+        if (currentItem.type === 'html') {
```

Comment:
Suggestion: The check for `html` should be placed in a dedicated function as it repeats multiple times. See also comment for `_handleCaptionEvents` and usage of `==`